### PR TITLE
#1200 request a cr reviewer schema change

### PR DESCRIPTION
**Pull request by callmejim1226**
## Changes

Updated schema.prisma so that the user has a list of change requests to review as a reviewer.

## Notes

Thanks for the help @Peyton-McKee and @RChandler234 on this haha

## Test Cases

Confirmed that the local app was running fine after this schema change and new migration.

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

**Review by Peyton-McKee — APPROVED** [review]
beautiful

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
Actually can you add this to the shared type as well

File: src/backend/src/transformers/change-requests.transformer.ts [hunk 1]
```diff
@@ -60,7 +60,8 @@ const changeRequestTransformer = (
     confirmDetails: changeRequest.activationChangeRequest?.confirmDetails ?? undefined,
     // stage gate cr fields
     leftoverBudget: changeRequest.stageGateChangeRequest?.leftoverBudget ?? undefined,
-    confirmDone: changeRequest.stageGateChangeRequest?.confirmDone ?? undefined
+    confirmDone: changeRequest.stageGateChangeRequest?.confirmDone ?? undefined,
+    requestedReviewers: changeRequest.requestedReviewers ?? undefined
```
Comment (Peyton-McKee) [inline src/backend/src/transformers/change-requests.transformer.ts:64]: ?? []

File: src/shared/src/types/user-types.ts [hunk 2]
```diff
@@ -39,6 +39,7 @@ export interface AuthenticatedUser {
   defaultTheme?: ThemeName;
   teamAsLeadId?: string;
   favoritedProjectsId?: number[];
+  changeRequestsToReviewId?: number[];
```
Comment (Peyton-McKee) [inline src/shared/src/types/user-types.ts:42]: way r these two optional they should just be empty lists

**Review by Peyton-McKee — COMMENTED** [review]
two optional things I think @RChandler234 might have different thoughts

File: src/shared/src/types/user-types.ts (hunk 2 above)
Comment (RChandler234) [reply src/shared/src/types/user-types.ts:42]: yeah empty lists would p be better

Comment (Peyton-McKee) [reply src/shared/src/types/user-types.ts:42]: we do this in a bunch of places rn and idk why there might be a reason

**Review by Peyton-McKee — APPROVED** [review]
lgtm

File: src/frontend/src/tests/test-support/test-data/authenticated-user.stub.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  firstName: 'Joe',
+  lastName: 'Shmoe',
+  email: 'shmoe.j@husky.neu.edu',
+  emailId: 'shmoe.j',
+  role: RoleEnum.ADMIN,
+  favoritedProjectsId: [],
+  changeRequestsToReviewId: []
+};
+
+export const exampleAllAuthenticatedUsers: AuthenticatedUser[] = [exampleAuthenticatedAdminUser];
```
Comment (Peyton-McKee) [inline src/frontend/src/tests/test-support/test-data/authenticated-user.stub.ts:19]: why do you need this

Comment (callmejim1226) [reply src/frontend/src/tests/test-support/test-data/authenticated-user.stub.ts:19]: @Peyton-McKee b/c the res function in test-utils.stub.ts line 90 requires AutehnticatedUser as an input. Previously, AuthenticatedUser had the same type with User but now it requires favoritedProjectsId and changeRequestsToReviewId fields to be defined. 

This is why I decided to add a new file, authenticated-user.stub.ts to define the example of authenticated user.

Comment (callmejim1226) [reply src/frontend/src/tests/test-support/test-data/authenticated-user.stub.ts:19]: Alternatively, I realized we can also use exampleAdminUser by adding favoritedProjectsId and changeRequestsToReviewId fields through spread syntax. 

-> {...exampleAdminUser, favoritedProjectsId: [], changeRequestsToReviewId: []} in test-utils.stub.ts, so we can delete authenticated-user.stub.ts if we want to.

Comment (callmejim1226) [reply src/frontend/src/tests/test-support/test-data/authenticated-user.stub.ts:19]: Lmk your thoughts on this. 

I need to push anyways to delete the comment in test-utils.stub.ts

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**